cpp/modmesh/view/RCameraController.cpp

```diff
@@ -0,0 +1,341 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <modmesh/view/RCameraController.hpp>
+
+#include <Qt3DInput/QKeyboardDevice>
+#include <Qt3DInput/QMouseDevice>
+#include <Qt3DRender/QCamera>
+
+namespace modmesh
+{
+
+RCameraInputListener::RCameraInputListener(
+    Qt3DInput::QKeyboardDevice * keyboardDevice,
+    Qt3DInput::QMouseDevice * mouseDevice,
+    callback_type callback,
+    QNode * parent)
+    : QEntity(parent)
+    , m_frame_action(new Qt3DLogic::QFrameAction)
+    , m_logical_device(new Qt3DInput::QLogicalDevice)
+    , m_keyboard_device(keyboardDevice)
+    , m_mouse_device(mouseDevice)
+    , m_callback(std::move(callback))
+    , m_rx_axis(new Qt3DInput::QAxis)
+    , m_ry_axis(new Qt3DInput::QAxis)
+    , m_tx_axis(new Qt3DInput::QAxis)
+    , m_ty_axis(new Qt3DInput::QAxis)
+    , m_tz_axis(new Qt3DInput::QAxis)
+    , m_left_mouse_button_action(new Qt3DInput::QAction)
+    , m_middle_mouse_button_action(new Qt3DInput::QAction)
+    , m_right_mouse_button_action(new Qt3DInput::QAction)
+    , m_shift_button_action(new Qt3DInput::QAction)
+    , m_alt_button_action(new Qt3DInput::QAction)
+    , m_left_mouse_button_input(new Qt3DInput::QActionInput)
+    , m_middle_mouse_button_input(new Qt3DInput::QActionInput)
+    , m_right_mouse_button_input(new Qt3DInput::QActionInput)
+    , m_shift_button_input(new Qt3DInput::QActionInput)
+    , m_alt_button_input(new Qt3DInput::QActionInput)
+    , m_mouse_rx_input(new Qt3DInput::QAnalogAxisInput)
+    , m_mouse_ry_input(new Qt3DInput::QAnalogAxisInput)
+    , m_mouse_tz_x_input(new Qt3DInput::QAnalogAxisInput)
+    , m_mouse_tz_y_input(new Qt3DInput::QAnalogAxisInput)
+    , m_keyboard_tx_pos_input(new Qt3DInput::QButtonAxisInput)
+    , m_keyboard_ty_pos_input(new Qt3DInput::QButtonAxisInput)
+    , m_keyboard_tz_pos_input(new Qt3DInput::QButtonAxisInput)
+    , m_keyboard_tx_neg_input(new Qt3DInput::QButtonAxisInput)
+    , m_keyboard_ty_neg_input(new Qt3DInput::QButtonAxisInput)
+    , m_keyboard_tz_neg_input(new Qt3DInput::QButtonAxisInput)
+{
+    init();
+
+    connect(
+        m_frame_action,
+        &Qt3DLogic::QFrameAction::triggered,
+        this,
+        [this](const float dt)
```

Comment:
You don't really need the `const` for the argument of fundamental types.  It usually does more harm than good.  But it's fine at this point.